lte/gateway/python/magma/pipelined/app/inout.py

```diff
@@ -276,45 +282,39 @@ def _get_gw_mac_address(self, gw_ip: str) -> str:
                        promisc=0)
 
             if res is not None:
-                self.logger.debug("resp: %s ", res.summary())
+                self.logger.debug("ARP Res pkt %s", res.show(dump=True))
                 mac = res[ARP].hwsrc
                 return mac
             else:
                 self.logger.debug("Got Null response")
+                return ""
 
         except Scapy_Exception as ex:
             self.logger.warning("Error in probing Mac address: err %s", ex)
-            return None
-
-    def _monitor_and_update(self, datapath):
-        current_feq = self.config.non_mat_gw_probe_frequency
-        if self._dhcp_gw_info.getMac() is not None:
-            latest_mac_addr = self._dhcp_gw_info.getMac()
-            self._install_default_egress_flows(datapath, latest_mac_addr)
-            flows.set_barrier(datapath)
+            return ""
+        except ValueError:
+            self.logger.warning("Invalid GW Ip address: [%s]", ip)
+            return ""
 
+    def _monitor_and_update(self):
         while True:
-            ip = self._dhcp_gw_info.getIP()
-            if ip is not None:
-                self.logger.info("GW found: %s", ip)
-                latest_mac_addr = self._get_gw_mac_address(ip)
-                if latest_mac_addr is not None:
-                    # go back to configured frequency.
-                    current_feq = self.config.non_mat_gw_probe_frequency
-                    if self._current_upstream_mac != latest_mac_addr:
-                        self._install_default_egress_flows(datapath, latest_mac_addr)
-                        flows.set_barrier(datapath)
-                        self._dhcp_gw_info.update_mac(latest_mac_addr)
+            updated_info = get_mobilityd_gw_info()
+            if updated_info:
+                gw_info = updated_info
+            if gw_info and gw_info.ip:
+                latest_mac_addr = self._get_gw_mac_address(gw_info.ip)
+                if len(latest_mac_addr) == 17 and \
+                        self._current_upstream_mac != latest_mac_addr:
+                    self._install_default_egress_flows(self._datapath, latest_mac_addr)
+                    gw_info.mac = latest_mac_addr
+                    set_mobilityd_gw_info(gw_info)
```

Comment:
nit: instead of mutating, construct a new GWInfo object here. Generally prefer immutability as a style thing, even if it means some extra object allocations